gpiozero/pins/__init__.py

```diff
@@ -6,39 +8,133 @@
     )
 str = type('')
 
-import io
-
-from .data import pi_info
 from ..exc import (
     PinInvalidFunction,
     PinSetInput,
     PinFixedPull,
+    PinSPIUnsupported,
     PinPWMUnsupported,
     PinEdgeDetectUnsupported,
+    SPIFixedClockMode,
+    SPIFixedBitOrder,
+    SPIFixedSelect,
+    SPIFixedWordSize,
     )
 
 
-PINS_CLEANUP = []
-def _pins_shutdown():
-    for routine in PINS_CLEANUP:
-        routine()
+class Factory(object):
+    """
+    Generates pins, SPI, and I2C interfaces for devices. This is an abstract
+    base class for pin factories. Descendents must override:
+
+    * :meth:`_get_address`
+    * :meth:`pin_address`
```

Comment:
I should probably word things a bit more clearly, but I think the lists are okay as they are - basically it's a mandatory override list followed by an optional override list (so implementers know the stuff they should tackle first to get something basic working, and what they can then explore to expand their implementation from there). I'll tweak the prose a bit to make it a little clearer.